src/main/scala/ignition/core/http/AsyncHttpClientStreamApi.scala

```diff
@@ -51,10 +52,9 @@ object AsyncHttpClientStreamApi {
                      headers: List[HttpHeader] = List.empty,
                      requestConfiguration: Option[RequestConfiguration] = None) {
 
```

Comment:
Maybe instead of returning try, we could make an assertion on creation of Request.
Something like: `require(URLUtils.parseUri(url).isSuccess)`